### PR TITLE
MH-12825: Add markdownlint to Travis CI

### PR DESCRIPTION
**Pull request by ziegenberg**
First attempt

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -39,4 +40,5 @@ script:
   - cd docs/guides/admin && mkdocs build && cd -
   - cd docs/guides/developer && mkdocs build && cd -
   - cd docs/guides/user && mkdocs build && cd -
+  - cd docs $$ grunt && cd -
```
Comment (gregorydlogan) [inline .travis.yml]: I'm guessing this should be docs *&&* grunt?

Comment (ziegenberg) [reply .travis.yml]: yes, you are right.

File: docs/Gruntfile.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+module.exports = function (grunt) {
```
Comment (lkiesow) [inline docs/Gruntfile.js]: Can we move these files into `guides` or even a `.` subdirectory? That's where the Markdown lives after all. Also, while it's a bit of nitpicking: two space indention and newline character at the end of file ;-P

Comment (ziegenberg) [reply docs/Gruntfile.js]: i'll move it

File: docs/guides/Gruntfile.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+          config: {
+            // configure the linting rules
+            default: false, // in the beginning disable all rules and enable them on a one-by-one basis
+            MD004: { // unordered list style
+              style: "consistent"
+            },
+            MD005: true, // inconsistent indentation for list items at the same level
+            MD006: true, // consider starting bulleted lists at the beginning of the line
+            MD007: { // unordered list indentation
+              indent: 2
```
Comment (lkiesow) [inline docs/guides/Gruntfile.js]: This shouldn't work. Afaik, markdown requires at least 4 spaces indentation for lists to be recognized at the next level

Comment (ziegenberg) [reply docs/guides/Gruntfile.js]: will fix

**Comment by ziegenberg** [comment]
that should be it for now. I guess you can merge/cherrypick whatever you like.

I'll continue with further rules in a new PR.

**Comment by ziegenberg** [comment]
can someone restart this build process? failing a build due to APT hick-ups on the CI environment is unfair.

**Comment by gregorydlogan** [comment]
I'm happy with this, looks good.  Are you deliberately aiming for develop rather than 5.x?  I'm happy either way, but the docs shouldn't have drifted much from 5.x.

**Comment by lkiesow** [comment]
I think `r/5.x` would actually be preferable. That way nobody merged stuff into 5.x which violates the rules which then breaks the builds once it's merged into `develop`.

@ziegenberg, while you bought the basic idea to list already, it would be great if you could give a very short overview about which rules are enforced at the technical meeting and also bring that to list as summary.

**Comment by gregorydlogan** [comment]
Either way, this is going to cause some conflicts with #173 among other docs PRs.  The further back we can push this the better imo.

**Comment by gregorydlogan** [comment]
Per discussion on IRC, blocking this until #173 is in.

**Comment by staubesv** [comment]
Hi @gregorydlogan, @lkiesow and @ziegenberg : #173 has been merged into r/5.x (and I've just merged r/5.x back to develop). This has caused the merge conflicts in this PR.

As I plan to work on some more documentation updates on both r/5.x and develop, how do we proceed? Will this one be merged into r/5.x or develop? Is there an ETA for this PR being merged?

**Comment by gregorydlogan** [comment]
As far as I'm concerned this is ready to go, aside from the conflicts.  As soon as the conflicts are resolved I will merge it in.

**Comment by ziegenberg** [comment]
So should I revert the last two commits to base it on r/5.x again?

**Comment by ziegenberg** [comment]
I have updated [the corresponding ticket ](https://opencast.jira.com/browse/MH-12825) with a listing of all the active rules and their explanation (and the reason when I felt the necessity)

**Pull request merged**